### PR TITLE
fix: add missing type stubs for `xml.dom.minidom.DOMImplementation`, where possible

### PR DESCRIPTION
**Pull request by kkirsche**
This adds additional details to `xml.dom.minidom.DOMImplementation` based on what I am able to determine from:
https://github.com/python/cpython/blob/main/Lib/xml/dom/minidom.py

This begins to address #6886

### CODE REVIEW

File: stdlib/xml/dom/minidom.pyi [hunk 1]
```diff
@@ -265,10 +265,10 @@ class Notation(Identified, Childless, Node):
     def __init__(self, name, publicId, systemId) -> None: ...
 
 class DOMImplementation(DOMImplementationLS):
-    def hasFeature(self, feature, version) -> bool: ...
-    def createDocument(self, namespaceURI: str | None, qualifiedName: str | None, doctype): ...
-    def createDocumentType(self, qualifiedName: str | None, publicId, systemId): ...
-    def getInterface(self, feature): ...
+    def hasFeature(self, feature: str, version: str | None) -> bool: ...
+    def createDocument(self, namespaceURI: str | None, qualifiedName: str | None, doctype: DocumentType | None) -> Document: ...
```
Comment (AlexWaygood) [inline stdlib/xml/dom/minidom.pyi:269]: I couldn't find definitive evidence for this in the source code, but the [documentation for this method](https://docs.python.org/3/library/xml.dom.html#xml.dom.DOMImplementation.createDocument) confirms this type annotation. Looks good.

File: stdlib/xml/dom/minidom.pyi [hunk 2]
```diff
@@ -265,10 +265,10 @@ class Notation(Identified, Childless, Node):
...
+    def createDocumentType(self, qualifiedName: str | None, publicId: Incomplete, systemId: Incomplete) -> Incomplete: ...
```
Comment (AlexWaygood) [inline stdlib/xml/dom/minidom.pyi:270]: According to the documentation for this function, `publicId` and `systemId` should both be strings: https://docs.python.org/3/library/xml.dom.html#xml.dom.DOMImplementation.createDocumentType

And we can see from the source that this function returns an instance of `DocumentType`: https://github.com/python/cpython/blob/e0d54a4a799dae4ebdd72a16bcf287ed62ae2972/Lib/xml/dom/minidom.py#L1477

```suggestion
    def createDocumentType(self, qualifiedName: str | None, publicId: str, systemId: str) -> DocumentType: ...
```

Comment (kkirsche) [reply stdlib/xml/dom/minidom.pyi:270]: Ah, I must have missed that. Thanks for catching this 👍

File: stdlib/xml/dom/minidom.pyi (hunk 1 above)
Comment (kkirsche) [reply stdlib/xml/dom/minidom.pyi:269]: I had been trying to use https://cs.github.com/?scopeName=All+repos&scope=&q=repo%3Apython%2Fcpython+path%3ALib%2Fxml%2Fdom+%22.createDocument%28%22 to work backwards through what calls were being used and comparing it to the documentation as my way of trying to find evidence, if that helps

**Review by AlexWaygood — APPROVED** [review]
Nice! Always great to get better types for our `xml` stubs — I think we probably all agree it's the weakest area in our stdlib stubs at the moment :)

File: stdlib/xml/dom/minidom.pyi (hunk 1 above)
Comment (AlexWaygood) [reply stdlib/xml/dom/minidom.pyi:269]: Oh, that is helpful, thanks! Especially with `xml`, it's sometimes pretty difficult to figure out the types just by studying the source code, so hints as to how you figured out non-obvious types are definitely appreciated :)

**Comment by kkirsche** [comment]
> Nice! Always great to get better types for our `xml` stub


For whatever reason, security standards like OpenSCAP often default to XML standards so I end up in either `xml` or [xsdata](https://xsdata.readthedocs.io/en/latest/) more often than I'd like. I'll need to look at what's there and what's missing to see what's involved with bringing it more inline with other areas of the standard library, in terms of coverage.

**Comment by AlexWaygood** [comment]
You can see here that there are a number of `xml` files that we have to exclude from the stricter pyright settings in our CI:

https://github.com/python/typeshed/blob/fb62ccabf5dfdc1a9d0cda145014cb585c77d59c/pyrightconfig.stricter.json#L9-L95

Ideally there wouldn't be any stdlib files on that list :)

**Comment by AlexWaygood** [comment]
> Ideally there wouldn't be any stdlib files on that list :)

(Having said that, I wouldn't bother with trying to fix the `distutils` or `lib2to3` entries. That feels like it would be somewhat of a waste of time, since those modules are deprecated and slated for removal soon :)

**Pull request merged**